File: source/downloads/code/LeetCode/Python/Missing-Number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        xor = 0
        for i in range(len(nums)):
            xor ^= i
            xor ^= nums[i]
        # because there is one missing
        return xor^len(nums)

File: source/downloads/code/LeetCode/Java/Missing-Number.java
public class Solution {
    public int missingNumber(int[] nums) {
        int xor = 0;
        int i;
        for (i=0;i<nums.length; i++){
            xor = xor ^ i;
            xor = xor ^ nums[i];
        }
        return xor ^ i;
    }
}
